components/pkg-export-kubernetes/src/bind.rs

```diff
@@ -27,6 +28,32 @@ pub struct Bind {
 impl FromStr for Bind {
     type Err = Error;
 
+    /// Create a `Bind` from a string.
+    ///
+    /// # Errors
+    ///
+    /// * `bind_spec` is not in the form `"name:service:group"`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use habitat_pkg_export_kubernetes::{Bind, Error};
+    ///
+    /// let b = "n:s:g".parse::<Bind>().unwrap();
+    ///
+    /// assert_eq!(b.name, "n");
+    /// assert_eq!(b.service, "s");
+    /// assert_eq!(b.group, "g");
+    ///
+    /// for s in ["ns:g", "nsg", "nsg", ":nsg:"].iter() {
+    ///     if let Err(e) = s.parse::<Bind>() {
+    ///         match e {
```

Comment:
Wouldn't it be easier to use assert instead of this match? Not sure how exactly this would look, but in pseudocode:

    assert_eq!(s.type(), MyError)

Reply:
that was what I wanted to do but the String inside the InvalidBind doesn't allow simple equality test. I asked on the IRC channel and this was the only way I was told is possible.